src/dummy/d2d1.h

```diff
@@ -1006,4 +1037,70 @@ struct dummy_ID2D1SolidColorBrush_tag {
 #define dummy_ID2D1SolidColorBrush_SetColor(self,a)             (self)->vtbl->SetColor(self,a)
 
 
+/*********************************************
+ ***  Interface ID2D1LinearGradientBrush   ***
+ *********************************************/
+
+typedef struct dummy_ID2D1LinearGradientBrushVtbl_tag dummy_ID2D1LinearGradientBrushVtbl;
+struct dummy_ID2D1LinearGradientBrushVtbl_tag {
+    /* IUnknown methods */
+    STDMETHOD(QueryInterface)(dummy_ID2D1LinearGradientBrush*, REFIID, void**);
+    STDMETHOD_(ULONG, AddRef)(dummy_ID2D1LinearGradientBrush*);
+    STDMETHOD_(ULONG, Release)(dummy_ID2D1LinearGradientBrush*);
+
+    /* ID2D1Resource methods */
+    STDMETHOD(dummy_GetFactory)(void);
+
+    /* ID2D1Brush methods */
+    STDMETHOD(dummy_SetOpacity)(void);
+    STDMETHOD(dummy_SetTransform)(void);
+    STDMETHOD(dummy_GetOpacity)(void);
+    STDMETHOD(dummy_GetTransform)(void);
+
+    /* ID2D1LinearGradientBrushBrush methods */
+    STDMETHOD(dummy_GetEndPoint)(void);
+    STDMETHOD(dummy_GetGradientStopCollection)(void);
+    STDMETHOD(dummy_GetStartPoint)(void);
+    STDMETHOD(dummy_SetEndPoint)(void);
+    STDMETHOD(dummy_SetStartPoint)(void);
+};
```

Comment:
These five are in some strange order. The order of the methods in the vtable **must** match the order in public SDK headers. If anyone starts using any such method in the future we could have hard-to-debug bugs.

Reply:
That is the order specified in the documentation: https://docs.microsoft.com/en-us/windows/desktop/api/d2d1/nn-d2d1-id2d1lineargradientbrush

Where do you get your information from?

Reply:
Documentation may be in any order, here likely alphabetical. But we have to follows the vtable layout in memory and that is described in SDK headers. Look into `<d2d1.h>` somewhere on your harddisk where you have SDK installed.